lib/screens/wods/components/add_wod_form.dart

```diff
@@ -94,44 +100,47 @@ class AddWodForm extends StatelessWidget {
           ),
           DividerMedium(),
           ReusableButton(
-              // onPressed: () {
-              //   bool validated = _formKey.currentState.validate();
-              //   if (validated) {
-              //     _formKey.currentState.save();
-              //     final data =
-              //         Map<String, dynamic>.from(_formKey.currentState.value);
-              //     context.read<WodRepository>().addWod(data);
-              //     Navigator.pop(context);
-              //   }
-              // },
               onPressed: () async {
                 bool validated = _formKey.currentState.validate();
                 if (validated) {
                   _formKey.currentState.save();
-                  final data =
-                      Map<String, dynamic>.from(_formKey.currentState.value);
-
-                  bool isSuccessful =
-                      await context.read<WodRepository>().addWods(data);
-                  if (isSuccessful) {
-                    Navigator.pop(context);
+                  if (widget.currentWod != null) {
+                    final data =
+                        Map<String, dynamic>.from(_formKey.currentState.value);
+                    bool isSuccesful2 = await context
+                        .read<WodRepository>()
+                        .updateWod(widget.currentWod.id, data);
+                    if (isSuccesful2) {
```

Comment:
You shouldn't name variables like this `something1`, `something2`, etc etc.

Variables names should be clean and descriptive. In this case, you can even use the same name `isSuccessful` on both sides of the if/else block because they will never be used at the same time: or it is initialized in the if, or in the else. Never on  both